pkg/oci/repository.go

```diff
@@ -225,6 +222,42 @@ func (r *Repository) pushBlob(layer v1.Layer) error {
 	return remote.WriteLayer(r.Repository, layer, r.remoteOpts...)
 }
 
+// PushStreamingImage pushes a reader to the repository as a streaming OCI image.
+// It accepts a media type and a byte slice as the blob.
+// Default media type is "application/vnd.oci.image.layer.v1.tar+gzip".
+// Annotations can be passed to the image manifest.
+func (r *Repository) PushStreamingImage(reference string, reader io.ReadCloser, mediaType string, annotations map[string]string) (string, error) {
```

Comment:
As an improvement this could return a `manifest` the same way we do with `FetchManifest`. User are more used to having a `manifest` returned when pushing an image.

Reply:
👍